common/types/offer.go

```diff
@@ -84,59 +59,6 @@ func (o *Offer) String() string {
 	)
 }
 
-// MarshalJSON is a custom JSON marshaller for Offer which enables serialisation of the private id field
-func (o Offer) MarshalJSON() ([]byte, error) {
-	return json.Marshal(&struct {
-		ID            string
-		Provides      ProvidesCoin
-		MinimumAmount float64
-		MaximumAmount float64
-		ExchangeRate  ExchangeRate
-		EthAsset      string
-	}{
-		ID:            o.id.String(),
-		Provides:      o.Provides,
-		MinimumAmount: o.MinimumAmount,
-		MaximumAmount: o.MaximumAmount,
-		ExchangeRate:  o.ExchangeRate,
-		EthAsset:      ethcommon.Address(o.EthAsset).Hex(),
-	})
-}
-
-// UnmarshalJSON is a custom JSON marshaller for Offer which enables deserialization of the private id field
-func (o *Offer) UnmarshalJSON(data []byte) error {
-	ou := &struct {
-		ID            string
-		Provides      ProvidesCoin
-		MinimumAmount float64
-		MaximumAmount float64
-		ExchangeRate  ExchangeRate
-		EthAsset      string
-	}{}
-	if err := json.Unmarshal(data, &ou); err != nil {
-		return err
-	}
-	id, err := hex.DecodeString(ou.ID)
-	if err != nil {
-		return fmt.Errorf("failed to unmarshal Offer ID err=%w", err)
-	}
-	if len(id) != len(o.id) {
-		return fmt.Errorf("offer ID has invalid length=%d", len(id))
-	}
-	copy(o.id[:], id)
-	o.Provides = ou.Provides
-	o.MinimumAmount = ou.MinimumAmount
-	o.MaximumAmount = ou.MaximumAmount
-	o.ExchangeRate = ou.ExchangeRate
-	if ou.EthAsset == "" {
-		// Default to EthAssetETH
-		o.EthAsset = EthAssetETH
-	} else {
-		o.EthAsset = EthAsset(ethcommon.HexToAddress(ou.EthAsset))
-	}
-	return nil
-}
```

Comment:
These functions were not moved, they were deleted. The ID field is now public and it has automatic JSON handling from the go-ethereum project.